BSP/STM32F413/Src/BSP_CAN.c

```diff
@@ -0,0 +1,402 @@
+    GPIO_InitStruct.GPIO_Speed = GPIO_Speed_50MHz;
+    GPIO_InitStruct.GPIO_OType = GPIO_OType_PP;
+    GPIO_InitStruct.GPIO_PuPd = GPIO_PuPd_UP;
+    GPIO_Init(GPIOA, &GPIO_InitStruct);
+
+    /* CAN configuration ********************************************************/  
+    /* Enable CAN clock */
+    RCC_APB1PeriphClockCmd(RCC_APB1Periph_CAN1, ENABLE);
+
+    /* CAN register init */
+    //CAN_DeInit(CAN1);
+
+    /* CAN cell init */
+    CAN_InitStruct.CAN_TTCM = DISABLE;
+    CAN_InitStruct.CAN_ABOM = DISABLE;
+    CAN_InitStruct.CAN_AWUM = DISABLE;
+    CAN_InitStruct.CAN_NART = DISABLE;
+    CAN_InitStruct.CAN_RFLM = DISABLE;
+    CAN_InitStruct.CAN_TXFP = DISABLE;
+    CAN_InitStruct.CAN_Mode = CAN_Mode_LoopBack; // TODO: change back to Normal after testing
+    CAN_InitStruct.CAN_SJW = CAN_SJW_1tq;
+
+    /* CAN Baudrate = 125 KBps
+        * 1/(prescalar + (prescalar*BS1) + (prescalar*BS2)) * Clk = CAN Baudrate
+        * The CAN clk is currently set to 20MHz (APB1 clock set to 20MHz in BSP_PLL_Init())
+    */
+    CAN_InitStruct.CAN_BS1 = CAN_BS1_3tq;
+    CAN_InitStruct.CAN_BS2 = CAN_BS2_4tq;
+    CAN_InitStruct.CAN_Prescaler = 16;
+    CAN_Init(CAN1, &CAN_InitStruct);
+
+    /* CAN filter init */
+    CAN_FilterInitStruct.CAN_FilterNumber = 0;
+    CAN_FilterInitStruct.CAN_FilterMode = CAN_FilterMode_IdMask;
+    CAN_FilterInitStruct.CAN_FilterScale = CAN_FilterScale_32bit;
+    CAN_FilterInitStruct.CAN_FilterIdHigh = 0x0000;
+    CAN_FilterInitStruct.CAN_FilterIdLow = 0x0000;
+    CAN_FilterInitStruct.CAN_FilterMaskIdHigh = 0x0000;
+    CAN_FilterInitStruct.CAN_FilterMaskIdLow = 0x0000;
+    CAN_FilterInitStruct.CAN_FilterFIFOAssignment = 0;
+    CAN_FilterInitStruct.CAN_FilterActivation = ENABLE;
+    CAN_FilterInit(CAN1, &CAN_FilterInitStruct);
+
+    //CAN_SlaveStartBank(CAN1, 0);
+
+    /* Transmit Structure preparation */
+    gTxMessage[0].ExtId = 0x5;
+    gTxMessage[0].RTR = CAN_RTR_DATA;
+    gTxMessage[0].IDE = CAN_ID_STD;
+    gTxMessage[0].DLC = 1;
+
+    /* Receive Structure preparation */
+    gRxMessage[0].StdId = 0x00;
+    gRxMessage[0].ExtId = 0x00;
+    gRxMessage[0].IDE = CAN_ID_STD;
+    gRxMessage[0].DLC = 0;
+    gRxMessage[0].FMI = 0;
+
+    /* Enable FIFO 0 message pending Interrupt */
+    CAN_ITConfig(CAN1, CAN_IT_FMP0, ENABLE);
+
+    // Enable Rx interrupts
+    NVIC_InitStruct.NVIC_IRQChannel = CAN1_RX0_IRQn;
+    NVIC_InitStruct.NVIC_IRQChannelPreemptionPriority = 0x00;
+    NVIC_InitStruct.NVIC_IRQChannelSubPriority = 0x00;
+    NVIC_InitStruct.NVIC_IRQChannelCmd = ENABLE;
+    NVIC_Init(&NVIC_InitStruct);	
+
+    /*if(NULL != txEnd) {
+        // Enable Tx Interrupts
+        NVIC_InitStruct.NVIC_IRQChannelPreemptionPriority = 0x0; // TODO: assess both of these priority settings
+        NVIC_InitStruct.NVIC_IRQChannelSubPriority = 0x0;
+        NVIC_InitStruct.NVIC_IRQChannelCmd = ENABLE;
+        NVIC_InitStruct.NVIC_IRQChannel = CAN1_TX_IRQn;
+        NVIC_Init(&NVIC_InitStruct);
+    }*/
```

Comment:
Are you planning on using tx interrupts? If not, are you planning on removing this?

Reply:
@Cam0Cow @ErickCortez98  Is there a reason this is commented out? I'm very sure that I pass in a TXEnd handler on the driver layer, is the transmit ISR initialized elsewhere?